---
title: "Lung Metagenomic Analysis"
author: "Chen, Zhibin"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r message=FALSE, warning=FALSE}
# load libraries that we need
library(metagenomeSeq)
library(biomformat)
```


```{r}
biom_file <- system.file("extdata", "min_sparse_otu_table.biom",
                         package = "biomformat")
b <- read_biom(biom_file)
# read in a biom file
biom2MRexperiment(b)
```

```{r}
data(mouseData)
b <- MRexperiment2biom(mouseData)
write_biom(b, biom_file = "/Users/courtneychen/BINF6309/module9otu_table.biom")
# the example of writing a MRexperiment object out as a BIOM file
```

```{r}
# loading count data
dataDirectory <- system.file("extdata", package = "metagenomeSeq") 
lung = loadMeta(file.path(dataDirectory, "CHK_NAME.otus.count.csv")) 
dim(lung$counts)
```

```{r}
# loading taxonomy
taxa = read.delim(file.path(dataDirectory, 
                            "CHK_otus.taxonomy.csv"), stringsAsFactors = FALSE)
```

```{r}
# loading metadata
clin = loadPhenoData(file.path(dataDirectory, "CHK_clinical.csv"), tran = TRUE)
ord = match(colnames(lung$counts), rownames(clin)) 
clin = clin[ord, ]
head(clin[1:2, ])
```

```{r}
# Creating a MRexperiment object
phenotypeData = AnnotatedDataFrame(clin)
phenotypeData
OTUdata = AnnotatedDataFrame(taxa)
OTUdata
obj = newMRexperiment(lung$counts,phenoData=phenotypeData,featureData=OTUdata)
obj
```

```{r}
# Example datasets
data(lungData)
lungData
data(mouseData)
mouseData
```

```{r}
# phenotype information can be accessed with phenoData & pData method
phenoData(obj)
head(pData(obj),3)
```

```{r}
# featureData and fData methods can access feature information
featureData(obj)
head(fData(obj)[,-c(2,10),3])
```

```{r}
# MRcounts function can acess row or normalized counts matrix
head(MRcounts(obj[, 1:2]))
```

```{r}
# A MRexperiment-class object can be easily subsetted
featuresToKeep = which(rowSums(obj) >= 100)
samplesToKeep = which(pData(obj)$SmokingStatus == "Smoker") 
obj_smokers = obj[featuresToKeep, samplesToKeep]
obj_smokers
```

```{r}
head(pData(obj_smokers), 3)
```

```{r}
head(normFactors(obj))
```

```{r}
normFactors(obj) <- rnorm(ncol(obj))
head(normFactors(obj))
```

```{r}
# libSize can access or replace library sizes / seq depths
head(libSize(obj))
```

```{r}
libSize(obj) <- rnorm(ncol(obj))
head(libSize(obj))
```

```{r}
# data can be filtered to maintain a threshold of min depth or OTU presence
data(mouseData)
filterData(mouseData, present = 10, depth = 1000)
```

```{r}
# mergeMRexperiments function can merge two MRexperiment-class objects
data(mouseData)
newobj = mergeMRexperiments(mouseData, mouseData)
```

```{r}
# calculating normalization factors
data(lungData)
p = cumNormStatFast(lungData)
# to calculate the scaling factors using cumNorm
lungData = cumNorm(lungData, p = p)
```

```{r}
# calculating normalization factors using Wrench
condition = mouseData$diet
mouseData = wrenchNorm(mouseData, condition = condition)
```

```{r}
# exporting the normalized count matrics
 mat = MRcounts(lungData, norm = TRUE, log = TRUE)[1:5, 1:5] 
exportMat(mat, file = file.path(dataDirectory, "tmp.tsv"))
```

```{r}
# to save sample statistics
exportStats(lungData[, 1:5], file = file.path(dataDirectory, "tmp.tsv"))
# default value being used warning
head(read.csv(file = file.path(dataDirectory, "tmp.tsv"), sep = "\t"))

```

```{r}
# statistical testing: comparing smoker's and non-smokers lung microbiome
data(lungData)
lungData = lungData[, -which(is.na(pData(lungData)$SmokingStatus))] 
lungData = filterData(lungData, present = 30, depth = 1)
lungData <- cumNorm(lungData, p = 0.5)
pd <- pData(lungData)
mod <- model.matrix(~1 + SmokingStatus, data = pd)
lungres1 = fitFeatureModel(lungData, mod)
# error in elnet_exp(ka, parm = alpha)
head(MRcoefs(lungres1))
```

```{r}
# using fitZig for differential abundance testing
data(lungData)
controls = grep("Extraction.Control", pData(lungData)$SampleType)
lungTrim = lungData[, -controls]
rareFeatures = which(rowSums(MRcounts(lungTrim) > 0) < 10)
lungTrim = lungTrim[-rareFeatures, ]
lungp = cumNormStat(lungTrim, pFlag = TRUE, main = "Trimmed lung data")
```

```{r}
lungTrim = cumNorm(lungTrim, p = lungp)
smokingStatus = pData(lungTrim)$SmokingStatus
bodySite = pData(lungTrim)$SampleType
normFactor = normFactors(lungTrim)
normFactor = log2(normFactor/median(normFactor) + 1)
mod = model.matrix(~smokingStatus + bodySite + normFactor)
settings = zigControl(maxit = 10, verbose = TRUE)
fit = fitZig(obj = lungTrim, mod = mod, useCSSoffset = FALSE, control = settings)
```

```{r}
# maxit=1 is for demonstration purposes
settings = zigControl(maxit = 1, verbose = FALSE)
mod = model.matrix(~bodySite)
colnames(mod) = levels(bodySite)
# fitting the ZIG model
res = fitZig(obj = lungTrim, mod = mod, control = settings) # The output of fitZig contains a list of various useful
# items. hint: names(res). Probably the most useful is the # limma 'MLArrayLM' object called fit.
zigFit = slot(res, "fit")
finalMod = slot(res, "fit")$design
contrast.matrix = makeContrasts(BAL.A - BAL.B, OW - PSB, levels = finalMod) 
fit2 = contrasts.fit(zigFit, contrast.matrix)
fit2 = eBayes(fit2)
topTable(fit2)
```

```{r}
# exporting fits
taxa = sapply(strsplit(as.character(fData(lungTrim)$taxa), split = ";"), function(i) {
i[length(i)] })
head(MRcoefs(fit, taxa = taxa, coef = 2))
```

```{r}
# log normal permutation test
coeffOfInterest = 2
res = fitLogNormal(obj = lungTrim, mod = mod, useCSSoffset = FALSE,
    B = 10, coef = coeffOfInterest)
# extract p.values and adjust for multiple testing res$p # are the p-values calculated through permutation 
adjustedPvalues = p.adjust(res$p, method = "fdr")
# extract the absolute fold-change estimates
foldChange = abs(res$fit$coef[, coeffOfInterest])
# determine features still significant and order by the
sigList = which(adjustedPvalues <= 0.05) 
sigList = sigList[order(foldChange[sigList])]
# view the top taxa associated with the coefficient of # interest.
head(taxa[sigList])
```

```{r}
classes = pData(mouseData)$diet
res = fitPA(mouseData[1:5, ], cl = classes)
# Warning - the p-value is calculating 1 despite a high # odd's ratio.
head(res)
```

```{r}
# feature correlations
cors = correlationTest(mouseData[55:60, ], norm = FALSE, log = FALSE) 
head(cors)
```

```{r}
# Unique OTUs or features
cl = pData(mouseData)[["diet"]]
uniqueFeatures(mouseData, cl, nsamples = 10, nreads = 100)
```

```{r}
# Aggregating counts
obj = aggTax(mouseData, lvl = "phylum", out = "matrix") 
head(obj[1:5, 1:5])
```

```{r}
obj = aggSamp(mouseData, fct = "mouseID", out = "matrix") 
head(obj[1:5, 1:5])
```

```{r}
# structural overview
trials = pData(mouseData)$diet
heatmapColColors = brewer.pal(12, "Set3")[as.integer(factor(trials))] 
heatmapCols = colorRampPalette(brewer.pal(9, "RdBu"))(50)
```

```{r}
# plotMRheatmap
plotMRheatmap(obj = mouseData, n = 200, cexRow = 0.4, cexCol = 0.4, trace = "none", col = heatmapCols, ColSideColors = heatmapColColors)
```

```{r}
cl = factor(pData(mouseData)$diet)
# plotOrd - can load vegan and set distfun = vegdist and
# use dist.method='bray'
plotOrd(mouseData, tran = TRUE, usePCA = FALSE, useDist = TRUE, bg = cl, pch = 21)
# plotRare
res = plotRare(mouseData, cl = cl, pch = 21, bg = cl)
# Linear fits for plotRare / legend
tmp = lapply(levels(cl), function(lv) lm(res[, "ident"]  ~ res[, "libSize"] - 1, subset = cl == lv))
for (i in 1:length(levels(cl))) { abline(tmp[[i]], col = i)
}
legend("topleft", c("Diet 1", "Diet 2"), text.col = c(1, 2), box.col = NA)
```

```{r}
head(MRtable(fit, coef = 2, taxa = 1:length(fData(lungTrim)$taxa)))
```

```{r}
patients = sapply(strsplit(rownames(pData(lungTrim)), split = "_"), function(i) {
i[3] })
pData(lungTrim)$patients = patients
classIndex = list(smoker = which(pData(lungTrim)$SmokingStatus ==
"Smoker"))
classIndex$nonsmoker = which(pData(lungTrim)$SmokingStatus ==
    "NonSmoker")
otu = 779
# plotOTU
plotOTU(lungTrim, otu = otu, classIndex, main = "Neisseria meningitidis")
# Now multiple OTUs annotated similarly
x = fData(lungTrim)$taxa[otu]
otulist = grep(x, fData(lungTrim)$taxa)
# plotGenus
plotGenus(lungTrim, otulist, classIndex, labs = FALSE, main = "Neisseria meningit ")
lablist <- c("S", "NS")
axis(1, at = seq(1, 6, by = 1), labels = rep(lablist, times = 3))
```

```{r}
classIndex = list(Western = which(pData(mouseData)$diet == "Western"))
classIndex$BK = which(pData(mouseData)$diet == "BK")
otuIndex = 8770
# par(mfrow=c(1,2))
dates = pData(mouseData)$date
plotFeature(mouseData, norm = FALSE, log = FALSE, otuIndex, classIndex,
    col = dates, sortby = dates, ylab = "Raw reads")
```

```{r}
# citing metagenomeSeq
citation("metagenomeSeq")
```

