---
title: "R Notebook"
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

```{r}
#if (!require("BiocManager"))
#    install.packages("BiocManager")
#BiocManager::install("maftools")
library(maftools)
```

```{r}
#path to TCGA LAML MAF file
laml.maf = system.file('extdata', 'tcga_laml.maf.gz', package = 'maftools') 
#clinical information containing survival information and histology. This is optional
laml.clin = system.file('extdata', 'tcga_laml_annot.tsv', package = 'maftools') 
laml = read.maf(maf = laml.maf, clinicalData = laml.clin)
```


```{r}
#Typing laml shows basic summary of MAF file.
laml
```


```{r}
#Shows sample summry.
getSampleSummary(laml)
#Shows gene summary.
getGeneSummary(laml)
#shows clinical data associated with samples
getClinicalData(laml)
#Shows all fields in MAF
getFields(laml)
#Writes maf summary to an output file with basename laml.
write.mafSummary(maf = laml, basename = 'laml')
```


```{r}
# plot the summary of the maf file
plotmafSummary(maf = laml, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```


```{r}
#oncoplot for top ten mutated genes.
oncoplot(maf = laml, top = 10)
```


```{r}
# the boxplot of the summerized data
laml.titv = titv(maf = laml, plot = FALSE, useSyn = TRUE)
#plot titv summary
plotTiTv(res = laml.titv)
```


```{r}
#lollipop plot for DNMT3A, which is one of the most frequent mutated gene in Leukemia.
# lollipop plots for amino acid changes
# by default, lollipop uses the longest isoform of the gene
lollipopPlot(
  maf = laml,
  gene = 'DNMT3A',
  AACol = 'Protein_Change',
  showMutationRate = TRUE,
  labelPos = 882
)
```


```{r}
# plotProtein function can plot the general protein domains
# plotProtein(gene = "TP53", refSeqID = "NM_000546")
# but it does not work with my working environemnt for some reason
```


```{r}
brca <- system.file("extdata", "brca.maf.gz", package = "maftools")
brca = read.maf(maf = brca, verbose = FALSE)
```

```{r}
rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.4)
```

```{r}
laml.mutload = tcgaCompare(maf = laml, cohortName = 'Example-LAML', logscale = TRUE, capture_size = 50)
```

```{r}
# estimate clonal status of top mutated genes
plotVaf(maf = laml, vafCol = 'i_TumorVAF_WU')
```

```{r}
# all.lreading and summerizing gistic outout files
all.lesions <- system.file("extdata", "all_lesions.conf_99.txt", package = "maftools")
amp.genes <- system.file("extdata", "amp_genes.conf_99.txt", package = "maftools")
del.genes <- system.file("extdata", "del_genes.conf_99.txt", package = "maftools")
scores.gis <- system.file("extdata", "scores.gistic", package = "maftools")

laml.gistic = readGistic(gisticAllLesionsFile = all.lesions, gisticAmpGenesFile = amp.genes, gisticDelGenesFile = del.genes, gisticScoresFile = scores.gis, isTCGA = TRUE)
```

```{r}
#GISTIC object
laml.gistic
```

```{r}
# genome plot 
gisticChromPlot(gistic = laml.gistic, markBands = "all")
```

```{r}
# bubble plot
gisticBubblePlot(gistic = laml.gistic)
```

```{r}
# oncoplot: sort the matrix according to annotations
gisticOncoPlot(gistic = laml.gistic, clinicalData = getClinicalData(x = laml), clinicalFeatures = 'FAB_classification', sortByAnnotation = TRUE, top = 10)
```

```{r}
# visualizing CBS segments
tcga.ab.009.seg <- system.file("extdata", "TCGA.AB.3009.hg19.seg.txt", package = "maftools")
plotCBSsegments(cbsFile = tcga.ab.009.seg)
```

```{r}
#exclusive/co-occurance event analysis on top 10 mutated genes. 
somaticInteractions(maf = laml, top = 25, pvalue = c(0.05, 0.1))
```

```{r}
# detecting cancer driver genes based on positional clustering
laml.sig = oncodrive(maf = laml, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')
```

```{r}
head(laml.sig)
```

```{r}
# plot the result
plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE, labelSize = 0.5)
```

```{r}
# adding and summerizing pfam domains
laml.pfam = pfamDomains(maf = laml, AACol = 'Protein_Change', top = 10)
```

```{r}
#Protein summary (Printing first 7 columns for display convenience)
laml.pfam$proteinSummary[,1:7, with = FALSE]
```

```{r}
#Domain summary (Printing first 3 columns for display convenience)
laml.pfam$domainSummary[,1:3, with = FALSE]
```

```{r}
#Survival analysis based on grouping of DNMT3A mutation status
mafSurvival(maf = laml, genes = 'DNMT3A', time = 'days_to_last_followup', Status = 'Overall_Survival_Status', isTCGA = TRUE)
```

```{r}
#Using top 20 mutated genes to identify a set of genes (of size 2) to predict poor prognostic groups
# id set of genes results in poor survival
prog_geneset = survGroup(maf = laml, top = 20, geneSetSize = 2, time = "days_to_last_followup", Status = "Overall_Survival_Status", verbose = FALSE)
```

```{r}
print(prog_geneset)
```

```{r}
mafSurvGroup(maf = laml, geneSet = c("DNMT3A", "FLT3"), time = "days_to_last_followup", Status = "Overall_Survival_Status")
```

```{r}
#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)
```

```{r}
#Considering only genes which are mutated in at-least in 5 samples in one of the cohort to avoid bias due to genes mutated in single sample.
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)
```
```{r}
# forest plots for visulizing the results above
forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1)
```

```{r}
# an alternative way of displaying result by plotting two oncoplots side by side
genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL', m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)
```

```{r}
coBarplot(m1 = primary.apl, m2 = relapse.apl, m1Name = "Primary", m2Name = "Relapse")
```

```{r}
# show gene wise differences with lollipopPlots
lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", AACol1 = "amino_acid_change", AACol2 = "amino_acid_change", m1_name = "Primary", m2_name = "Relapse")
```

```{r}
# takes any clinical feature associated with the samples and performs enrichment anylysis
fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')
```

```{r}
#Results are returned as a list. Significant associations p-value < 0.05
fab.ce$groupwise_comparision[p_value < 0.05]
```

```{r}
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05, geneFontSize = 0.5, annoFontSize = 0.6)
```

```{r}
# check for drug-gene interaction and gene druggability
dgi = drugInteractions(maf = laml, fontSize = 0.75)
# the plot of potential druggable gene categories along with upto top 5 genes involved in them
```

```{r}
# results for known/reported drugs to interact with DNMT3A
dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)
```

```{r}
#Printing selected columns.
dnmt3a.dgi[,.(Gene, interaction_types, drug_name, drug_claim_name)]
```

```{r}
# check for enrichment of known Oncogenic Signaling Pathways in TCGA cohors
OncogenicPathways(maf = laml)
```

```{r}
# visualize complete pathway
PlotOncogenicPathways(maf = laml, pathways = "RTK-RAS")
```
